### PR TITLE
Add support for indirect calls to the DFS generator.

### PR DESCRIPTION
**Pull request by xyzsam**
When --use_indirect_calls is specified, the generator will emit
CodeBlocks that specify INDIRECT_CALL and a list of targets along with
call probabilities, rather than a sequence of conditional branches +
direct calls.

Fixes issue #6.

### CODE REVIEW

File: frontend/src/frontend/cfg_generator/dfs_chase_gen.py [hunk 1]
```diff
@@ -58,9 +74,43 @@ def _add_code_block_with_branch(
             block.terminator_branch.taken_probability.append(probability)
         return block
 
+    def _generate_indirect_call_code_block(
+            self, call_targets: List[int],
+            callee_probability: float) -> cfg_pb2.CodeBlock:
+        """Generates a single CodeBlock that indirectly calls N targets.
```
Comment (rpgolshan) [inline frontend/src/frontend/cfg_generator/dfs_chase_gen.py:80]: One of two targets and not N targets?

**Review by rpgolshan — APPROVED** [review]
This looks good. However the code_generator module needs to be modified to support multiple targets in an indirect call.

File: frontend/src/frontend/cfg_generator/dfs_chase_gen.py (hunk 1 above)
Comment (xyzsam) [reply frontend/src/frontend/cfg_generator/dfs_chase_gen.py]: Done.

**Comment by xyzsam** [comment]
> This looks good. However the code_generator module needs to be modified to support multiple targets in an indirect call.

Right. It should be fairly similar to the array of goto labels, except it's going to call a bunch of functions instead (so we need all the functions to have the same signature).

**Pull request merged**